planner/core/plan_cacheable_checker_test.go

```diff
@@ -244,3 +244,110 @@ func TestCacheable(t *testing.T) {
 	}
 	require.True(t, core.Cacheable(stmt, is))
 }
+
+func TestGeneralPlanCacheable(t *testing.T) {
+	store := testkit.CreateMockStore(t)
+
+	tk := testkit.NewTestKit(t, store)
+
+	tk.MustExec("use test")
+	tk.MustExec("create table t1(a int, b int) partition by range(a) ( partition p0 values less than (6), partition p1 values less than (11) )")
+	tk.MustExec("create table t2(a int, b int) partition by hash(a) partitions 11")
+	tk.MustExec("create table t3(a int, b int)")
+	tbl := &ast.TableName{Schema: model.NewCIStr("test"), Name: model.NewCIStr("t3")}
+	is := tk.Session().GetInfoSchema().(infoschema.InfoSchema)
+	var stmt ast.Node = &ast.ShowStmt{}
+	require.False(t, core.GeneralPlanCacheable(stmt, is))
+
+	stmt = &ast.LoadDataStmt{}
```

Comment:
Better to use queries as the input of this test instead of constructing `AST Node` directly:
```
q := "select * from t1, t2"
stmt := parser.New().ParseOneStmt(q)
require.False(t, core.GeneralPlanCacheable(stmt, is))
```